test_runner/regress/test_installed_extensions.py

```diff
@@ -85,3 +87,26 @@ def test_installed_extensions(neon_simple_env: NeonEnv):
             assert ext["n_databases"] == 2
             ext["versions"].sort()
             assert ext["versions"] == ["1.2", "1.3"]
+
+    # check that /metrics endpoint is available
+    # ensure that we see the metric before and after restart
+    res = client.metrics()
+    info("Metrics: %s", res)
+    m = parse_metrics(res)
+    neon_m = m.query_all("installed_extensions", {"extension_name": "neon", "versions": "1.2,1.3"})
+    assert len(neon_m) == 1
+    for sample in neon_m:
+        assert sample.value == 2
+
+    endpoint.stop()
+    endpoint.start()
+
+    time.sleep(1)
```

Comment:
I'm surprised we don't have a wait until start type function here.

Reply:
thank you for spotting this.
actually, we're not waiting for compute to start, but for `get_installed_extensions` which runs in a separate thread https://github.com/neondatabase/neon/pull/9375/files#diff-026c259a839e1a9bfe6c2a4e725431c4e8a48131b0c483bde0fb980bf7632e73R1124

I rewrote that part of the test to reflect this logic.